### ISSUE TITLE
Segmentation fault when trying to run the daemon

### ISSUE THREAD

**Issue by it-s** [AppImage/appimaged#59]
I'm getting a Segmentation fault when trying to run appimaged appimage under Elementary OS 5
Here is the console output:
```
.local/bin/appimaged -v
appimaged, continuous build (commit 1dd440a), build 88 built on 2019-01-25 18:18:24 UTC
AppImage type 2

(appimaged:11277): GLib-CRITICAL **: 16:59:08.058: g_file_test: assertion 'filename != NULL' failed
AppImage type 2
Segmentation fault
```

**Comment by probonopd** [comment]
Duplicate of https://github.com/AppImage/appimaged/issues/58?
Does it work with the test version linked there?

**Comment by tresf** [comment]
Same issue on Ubuntu 18.04, fresh install.  Test version fixed it.

**Comment by probonopd** [comment]
@tresf did you try the test version linked in #58 as well?

**Comment by probonopd** [comment]
Thanks for testing @tresf. Glad to hear that the test version fixes it.
@TheAssassin @azubieta I'd like to switch appimaged to libappimage 0.2 rather soon because it fixes this known issue.

**Comment by it-s** [comment]
> Duplicate of #58?
> Does it work with the test version linked there?

Yes this fixes the issue. Thanks!

**Comment by probonopd** [comment]
Closing here since it seems to be a duplicate of #58 indeed.

**Comment by TheAssassin** [comment]
I keep repeating myself on this: libappimage 0.2.0 is still far from ready for real world use. You can ask @azubieta, too, since you don't seem to believe me. We don't want to ship bananaware any more, hence the extensive review & refactoring.

**Comment by TheAssassin** [comment]
It solves one issue, but might create 1000 new ones. This issue is not entirely representative for the state of the project.

**Comment by azubieta** [comment]
To be clear, we don't know how many issues it will solve or create. All that we can guarantee is that it works for every tests case included (which are much more than the ones in 0.1). In terms of overall quality, I consider that it's superior. This doesn't mean that we must use it immediately everywhere as we cannot include in the tests cases every possible scenario. It would be wise to make a beta release of our tools and give it to loyal users in order to identify and correct any regression bug.

**Comment by probonopd** [comment]
Not saying that we should use 0.2 _everywhere now_, but seeing that it is already being used in projects, e.g., in https://github.com/azubieta/AppImageInfo/pull/17, and given that I have been running appimaged with libappimage 0.2 for weeks now on my main system with no apparent ill side effects compared to before, I am really considering to switch `appimaged` to it rather sooner than later.
